superset/reports/notifications/S3.py

```diff
@@ -0,0 +1,110 @@
+import json
+import logging
+from io import BytesIO
+from uuid import uuid4
+import datetime
+import boto3
+from flask_babel import gettext as __
+from superset import app
+from superset.exceptions import SupersetErrorsException
+from superset.reports.models import ReportRecipientType
+from superset.reports.notifications.base import BaseNotification
+from superset.reports.notifications.exceptions import NotificationError
+
+logger = logging.getLogger(__name__)
+
+class S3SubTypes:
+    """
+    Defines different types of AWS S3 configurations.
+    """
+    S3_CRED = 'AWS_S3_credentials'
+    S3_CONFIG = 'AWS_S3_pyconfig'
+    S3_ROLE = 'AWS_S3_IAM'
+
+
+class S3Notification(BaseNotification):
+
+    type = ReportRecipientType.S3
+    def _get_inline_files(self):
+        current_datetime = datetime.datetime.now()
+        formatted_date = current_datetime.strftime("%Y-%m-%d")
+        report_name= self._content.name
+        name_prefix=f"{report_name}/{formatted_date}/"
```

Comment:
Can we read this from a config prop? Folks might wanna tune the path on their own